bindings/python/pymongoarrow/__init__.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pymongoarrow.libbson.version import __version__ as libbson_version
+from pymongoarrow.version import __version__, _MIN_LIBBSON_VERSION
+
+
+try:
+    from pkg_resources import parse_version as _parse_version
+except ImportError:
+    from distutils.version import LooseVersion as _LooseVersion
+
+    def _parse_version(version):
+        return _LooseVersion(version)
+
+
+if _parse_version(libbson_version) < _parse_version(_MIN_LIBBSON_VERSION):
+    raise RuntimeError(
+        "Expected libbson version {} or greater, found {}}".format(
+            _MIN_LIBBSON_VERSION, libbson_version))
```

Comment:
I think using the library at import time like this can cause some problems down the line. Imagine when we want to integrate this library with pymongo. Then pymongo will look like this:

```
try:
    import pymongoarrow
except ImportError:
    pass  # not installed 
```

With the current design we need to catch RuntimeError as well which is a bit odd:
```
try:
    import pymongoarrow
except (ImportError, RuntimeError):
    pass  # not installed 
```

I suggest we defer calling into the library until it's first used. For example:
```
__version__ = bson_get_version().decode('utf-8')
```
Would become:
```
def libbson_version():
    return bson_get_version().decode('utf-8')
```

Feel free to open a follow up ticket if you like to merge this as is and discuss this issue further in Jira.

Edit: a simple alternative option might be to change this line to raise an ImportError instead of a RuntimeError.

Reply:
Good point except there is no one point of entry for loading libbson in this case so even if the version check is done lazily, it would have to be explicitly added to every possible method in the libbson bindings that could be directly called by external code.

I have changed it to raise an ImportError instead.